Handle case where one or both of results and control all NA

addison-grant opened the pull request:
It is apparently possible that this is the case. I don't know if this is the correct fix.

I ran into this when trying to rebuild the Bight '13 summary

addison-grant committed:
Handle case where one or both of results and control all NA

It is apparently possible that this is the case. I don't know if this is the correct fix.

addison-grant commented:
@r7butler, do you know if this is a good change or not, or should we run this change by Darrin?

r7butler commented:
That would be the correct way to handle that case

But lets add a warning message in that case, since that's something that really should not happen

`warning("For one or more batches, all replicates were missing values, for both the actual sample and the control. Please double check your data.")`

That being said, we need to double check the bight '13 data that caused this issue

Go ahead and send me that data filtered to the results toxbatch and the control batch that caused the problem

addison-grant committed:
Add warning for results/control all-NA

addison-grant committed:
Make corrections to summary output

1. Group by a few more columns (endpoint, matrix)
2. Fix QA Code grouping

addison-grant committed:
Make corrections to names

Based on conversation with Darrin.

addison-grant committed:
Fix repeating comments

Comments were getting repeated and collapsed on "; ".
This collapses properly.

addison-grant commented:
If the latest commits are good, then I think I need to fix code comments.

addison-grant committed:
Match pctcontrol name in published 2018 summary

addison-grant commented:
I checked the qacode output, and found that it isn't working as expected, seeing things like `AJ` instead of `A, J`. I found docs confirming that QA codes are all single-letter: 
<img width="599" alt="image" src="https://github.com/user-attachments/assets/aaee66c7-a800-4f3c-aef4-5ed4f923b1a1" />
https://ftp.sccwrp.org/pub/download/BIGHT18/Bight18SedQualityIMPlan.pdf

I'll update the code to reflect this information.

addison-grant committed:
Fix qacode splitting and rejoining

Before: had codes like "AJ" in the end
After: end up with codes like "A, J", for example

addison-grant commented:
There are a lot of `writelog` calls that are out of date with these changes, I believe. Do you generate those automatically, @r7butler?

addison-grant committed:
Mark some numerical column NAs with -88

Add some comments on functionality
Fix comment collapsing to remove NAs

addison-grant commented:
Some remaining issues

- [x] Should use `SC` / `NSC` for `sigeffect`
- [x] Fix units to be `resultunits` from Results data
- [x] Fix `fieldreps` to be `1` where they are currently `NA`
- [x] Drop SampleA, SampleB, etc., stations
- [x] Ask Paul about BOEMs stations (do we need to keep them?)
   - Paul thinks we probably just keep B##-#####, but to check with Robert

~~I think we'll do all these here, but I need to take a closer look at the input Results data. I am just wondering if we should preprocess the results data outside of `tox.summary` or have special logic in `tox.summary`.~~

I'm doing these in post-processing now, but let me know if you think we should do them in here, @r7butler.

addison-grant committed:
Fix grouping columns

Mistakenly altered these in 737021a582a0196f905b7f769a2a327662d8245d

Grouping by endpoint and matrix isn't required because they are unique to species.

addison-grant committed:
Remove stray character string

`tox_categories` is loaded by the presence of `LazyData: true` in the top-level DESCRIPTION file.

addison-grant committed:
Comment on potential source of bug

addison-grant committed:
Keep matrix column in summary

I believe this was the cause of the missing matrix column in the output

addison-grant commented:
I split this up into multiple PRs
#29
#25 
#30  
#31 
#32
